### PR TITLE
All comments

### PR DESCRIPTION
**Pull request by rich-nguyen**
Adding an option for users to load all comments. It's a paginated load, so requires a lot of async work.

### CODE REVIEW

File: common/app/conf/switches.scala [hunk 1]
```diff
@@ -152,6 +152,11 @@ object Switches {
     safeState = Off, sellByDate = never
   )
 
+  val DiscussionAllPageSizeSwitch = Switch("Performance", "discussion-all-page-size",
```
Comment (phamann) [inline common/app/conf/switches.scala]: IS this really a performance switch?

File: static/src/javascripts/projects/common/modules/discussion/whole-discussion.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+    WholeDiscussion.prototype.loadAllComments = function () {
+
+        // Always load the first page, to retrieve the number of comments in the discussion.
+        var queryParams = {
+            orderBy: this.params.orderBy,
+            page: 1,
+            pageSize: commentsPerPage,
+            displayThreaded: this.params.displayThreaded
+        };
```
Comment (phamann) [inline static/src/javascripts/projects/common/modules/discussion/whole-discussion.js]: This object is duplicated twice, I'd create a function to generate them which accepts page as an argument.

**Comment by phamann** [comment]
As discussed I don't like that this current implementation that forces all ajax requests to load on pageload, even if the user never scrolls down the comments. I'd prefer if we only load all comments if the user actually intents to viewing them, such as clicking the "View all comments +" button.

**Comment by phamann** [comment]
Let's chat with @cantlin when he is in.

**Comment by phamann** [comment]
:+1: Other than the comments, very nice/clean approach with the promise resolution etc.

File: static/src/javascripts/projects/common/modules/discussion/whole-discussion.js (hunk 2 above)
Comment (rich-nguyen) [inline static/src/javascripts/projects/common/modules/discussion/whole-discussion.js]: I did, I just noticed I left this one behind!

**Comment by rich-nguyen** [comment]
Today seems like the perfect day to refactor comments.

Some new "All page size" features -  always load comments in oldest order to avoid pages being offset by new comments, and defer the big load of comments until the user clicks "show more"

**Pull request merged**